pkg/istio-aux/util.go

```diff
@@ -64,18 +74,18 @@ func setAnnotation(meta *metav1.ObjectMeta, key string, value map[string]interfa
 		existing := map[string]interface{}{}
 		if err := yaml.Unmarshal([]byte(val), &existing); err == nil {
 			for k, v := range existing {
```

Comment:
The true issue is actually with this function, the assignment to value/localvalue should raise a compile error as it should be a read-only input, sure Go has something for this?

Then the implementation of the function should build it's own copy/map on the stack to merge the values before it's assigned back to the meta.annotation.

Reply:
> The true issue is actually with this function, the assignment to value/localvalue should raise a compile error as it should be a read-only input, sure Go has something for this?

No. Normally, you'd use `const` on the top level `IstioPodAnnotationValue`, but because it has an inline assignment, it cannot be `const`.